presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java

```diff
@@ -520,6 +474,22 @@ public Object validateClusteringKey(Object value)
         }
     }
 
+    public static boolean containsUnsupportedType(DataType dataType)
+    {
+        if (!dataType.isCollection()) {
```

Comment:
This could probably be simplified to (i also inverted meaning of the method):
```
public static boolean isFullySupported(DataType dataType)
{
    if (!toCassandraType(dataType.getName()).isPresent()) {
        return false;
    }

    return dataType.getTypeArguments().stream()
            .allMatch(CassandraType::isFullySupported);
}
```